src/levanter/models/flash_attention.py

```diff
@@ -307,6 +307,11 @@ def do_inner_block(state):
             dV_j = dV_j + hax.dot(QPos.name, p_ij, dO_i).astype(dV_j.dtype)
             dK_j = dK_j + hax.dot(QPos.name, dAttn_ij, q_i).astype(dK_j.dtype)
 
+            # GQA-specific: eliminate unnecessary axes (e.g. 'q_heads_per_group')
```

Comment:
I'm not sure if this is a good way to do this summation. If there's a better place, please let me know. I'm guessing the placement of this summation logic would affect performance. @dlwh